components/catalogs/front/src/components/ProductSelection/criteria/AttributeYesNoCriterion/AttributeYesNoCriterion.integration.tsx

```diff
@@ -0,0 +1,390 @@
+const changeLocaleTo = (locale: string) => changeSelectValueTo('locale', locale);
+const changeInputTo = (value: string) => changeSelectValueTo('value', value);
+const changeSelectValueTo = (selector: string, value: string) => {
+    const select = screen.getByTestId(selector);
+    fireEvent.click(within(select).getByRole('textbox'));
+    fireEvent.click(screen.getByText(value));
+};
+
+beforeEach(() => {
+    fetchMock.mockResponse(req => {
+        switch (req.url) {
+            // useAttribute
+            case '/rest/catalogs/attributes/name_localizable_scopable':
+                return Promise.resolve(
+                    JSON.stringify({
+                        label: 'Name',
+                        code: 'name',
+                        type: 'pim_catalog_boolean',
+                        scopable: true,
+                        localizable: true,
+                    })
+                );
+            case '/rest/catalogs/attributes/name_scopable':
+                return Promise.resolve(
+                    JSON.stringify({
+                        label: 'Name',
+                        code: 'name',
+                        type: 'pim_catalog_boolean',
+                        scopable: true,
+                        localizable: false,
+                    })
+                );
+            case '/rest/catalogs/attributes/name_localizable':
+                return Promise.resolve(
+                    JSON.stringify({
+                        label: 'Name',
+                        code: 'name',
+                        type: 'pim_catalog_boolean',
+                        scopable: false,
+                        localizable: true,
+                    })
+                );
+            // useChannel
+            case '/rest/catalogs/channels/ecommerce':
+                return Promise.resolve(JSON.stringify(channelEcommerce));
+            case '/rest/catalogs/channels/print':
+                return Promise.resolve(JSON.stringify(channelPrint));
+            // useChannelLocales
+            case '/rest/catalogs/channels/ecommerce/locales':
+                return Promise.resolve(JSON.stringify([localeUS, localeFR]));
+            // useLocales
+            case '/rest/catalogs/locales':
+                return Promise.resolve(JSON.stringify([localeUS, localeFR, localeDE]));
+            // useInfiniteChannels
+            case '/rest/catalogs/channels?page=1&limit=20':
+                return Promise.resolve(JSON.stringify([channelEcommerce, channelPrint]));
+            default:
+                throw Error(req.url);
+        }
+    });
+});
+
+test('it renders the scopable and localizable text attribute criterion', async () => {
+    render(
+        <ThemeProvider theme={pimTheme}>
+            <ReactQueryWrapper>
+                <AttributeYesNoCriterion
+                    state={{
+                        field: 'name_localizable_scopable',
```

Comment:
```suggestion
                        field: 'name_localizable_scopable',
```

It's not essential, but I would have picked another name for the attributes.
Having a boolean attribute called `name` or `name_localizable_scopable` does not really make sense.

Reply:
I would say that it's a matter of taste but I tend to be careful about having tests that make sense.

If you read your test aloud, it says, "I have an attribute `name`, I want to find all products where the value is `true`".

In AttributeSimpleSelectCriterion, I always named my attribute `clothing_size` with the options `xs, s, m, l, xl`.
In AttributeNumberCriterion, I named it `number_battery_cells`.

It's not essential as I said, but it make more sense and the test is easier to read, IMO.

Reply:
You are not only writing tests to check that your code is working, you are also writing the intent behind the behavior.
If somebody else goes back to this piece of code in several years, it will really confuse them why it's done like this.

Can you rename it to an attribute code that makes sense for a boolean ?

Reply:
It's renamed : display_color => Display Color :)